packages/wallets/saifu/src/adapter.ts

```diff
@@ -0,0 +1,270 @@
+                            wallet.off('connect', connect);
+                            resolve();
+                        };
+
+                        wallet._handleDisconnect = (...args: unknown[]) => {
+                            wallet.off('connect', connect);
+                            reject(new WalletWindowClosedError());
+                            return handleDisconnect.apply(wallet, args);
+                        };
+
+                        wallet.on('connect', connect);
+
+                        wallet.connect().catch((reason: any) => {
+                            wallet.off('connect', connect);
+                            reject(reason);
+                        });
+                    });
+                } catch (error: any) {
+                    if (error instanceof WalletError) throw error;
+                    throw new WalletConnectionError(error?.message, error);
+                } finally {
+                    wallet._handleDisconnect = handleDisconnect;
+                }
+            }
+
+            if (!wallet.publicKey) throw new WalletAccountError();
+
+            let publicKey: PublicKey;
+            try {
+                publicKey = new PublicKey(wallet.publicKey.toBytes());
+            } catch (error: any) {
+                throw new WalletPublicKeyError(error?.message, error);
+            }
+
+            wallet.on('disconnect', this._disconnected);
+
+            this._wallet = wallet;
+            this._publicKey = publicKey;
+
+            this.emit('connect', publicKey);
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        } finally {
+            this._connecting = false;
+        }
+    }
+
+    async disconnect(): Promise<void> {
+        const wallet = this._wallet;
+        if (wallet) {
+            wallet.off('disconnect', this._disconnected);
+
+            this._wallet = null;
+            this._publicKey = null;
+
+            try {
+                await wallet.disconnect();
+            } catch (error: any) {
+                this.emit('error', new WalletDisconnectionError(error?.message, error));
+            }
+        }
+
+        this.emit('disconnect');
+    }
+
+    async sendTransaction(
+        transaction: Transaction,
+        connection: Connection,
+        options?: SendTransactionOptions
+    ): Promise<TransactionSignature> {
+        try {
+            const wallet = this._wallet;
+            if (wallet && 'signAndSendTransaction' in wallet && !options?.signers) {
+                transaction.feePayer = transaction.feePayer || this.publicKey || undefined;
+                transaction.recentBlockhash =
+                    transaction.recentBlockhash || (await connection.getRecentBlockhash('finalized')).blockhash;
+
+                const { signature } = await wallet.signAndSendTransaction(transaction, options);
+                return signature;
+            }
```

Comment:
It will just use the one from BaseSignerWalletAdapter which works fine for Saifu as is, so there was no need to override it - https://github.com/solana-labs/wallet-adapter/blob/1ce2c5ac62abff8082a6c96612959dac499d2591/packages/core/base/src/signer.ts#L13:L52

Reply:
We used to at some point but it got removed to align with behavior of other wallets while this PR was open. If you recommend having it supported then I'll add it back with a fallback to `parent` if the method doesn't exist on the object, so we can transparently re-add it in the next Saifu release

Reply:
Pretty sure all the major wallets have this method? The base implementation is just to support older wallets that don't have it.

Reply:
Okay, I've re-added `sendTransaction` with fallback to base implementation when `signAndSendTransaction` isn't in the wallet object yet. We'll add it back in the next release 👍